### ISSUE TITLE
Wrong available disk space displayed

### ISSUE THREAD

**Issue by MightyCreak** [ValveSoftware/steam-for-linux#3881]
Hi!

I've moved my SteamApps directory to another place on my SSD drive, and then created a symbolic link to this directory, like this:

    cd ~/.local/share/Steam && ln -s /fastdrive/SteamApps

But now in Steam, when I want to install a new game, the available disk space is wrong. It shows the available disk space of `~/.local/share/Steam` instead of `~/.local/share/Steam/SteamApps`.

Here is the output of `df -h` for each directory:

    $ df -h ~/.local/share/Steam/
    Filesystem      Size  Used Avail Use% Mounted on
    /dev/sdb5       827G  501G  284G  64% /home

    $ df -h ~/.local/share/Steam/SteamApps 
    Filesystem                    Size  Used Avail Use% Mounted on
    /dev/mapper/fedora-fastdrive  172G   83G   81G  51% /fastdrive

I tried to chose a different directory in the dropdown list, but it says I don't have the rights, which is weird since the directory has more rights than `~/.local/share/Steam` (755 vs 705). But well, that's another story.

**Comment by jurf** [comment]
Yeah, encountered this too, Steam should really `df` _inside_ `steamapps`, not the parent directory, when installing games.

**Comment by jurf** [comment]
In my case there were no symlinks involved, but it’s the same issue, symlink or not.

**Comment by TechtonicSoftware** [comment]
For me I have 100GB less than is available (not sure if this is an intentional safety mechinism. No symlinks other than the default created one for $HOME/.steam/steam -> $HOME/.local/share/Steam ..

**Comment by har7an** [comment]
Can we please have an option to entirely disable or ignore this up-front free space check? I deliberately chose to mount `steamapps/common` and `steamapps/downloading` from a separate drive, because that uses another FS than my main host with different mount options set. I do *not* want the other subfolders from `steamapps/` on that same disk though, because I'd like to snapshot these along with my hosts rootfs so I don't lose savegames any more (stored in `steamapps/compatdata`, for example) because in the past the cloud sync has deleted a bunch of files I held very dear in fact.

Given the amount of modern file systems out there (XFS, Btrfs, ZFS, you name it...), I feel like the assumptions you're making about free space being available don't hold any more. If `df` tells me there are *x* GB available, there may well be more space in reality due to compression or reflinks (for example).

I think it'd be fair if you warned users that their storage may be insufficient, but outright disabling installation in that case is too harsh IMO.
